gen_docker_compose.py

```diff
@@ -79,6 +79,8 @@ def generate_compose_file(
         for k, v in self.config['versions'].items():
             template_args[k] = v
 
+        template_args["FIRMWARE_SHORT_HASH"] = os.getenv("FIRMWARE_SHORT_HASH", "unknown")
```

Comment:
This seems counter intuitive. We're reading some args from `settings.ini` and others are read from ENV within the code. Because of this just looking at `settings.ini` we cannot determine that args available to a template.

This could be implemented in a better way using a couple of approaches.

### 1. Include `FIRMWARE_SHORT_HASH` in `settings.ini` with a special prefix like $.

For example

```
[versions]
FIRMWARE_VERSION=2022.01.12.1-2
FIRMWARE_SHORT_HASH=$FIRMWARE_SHORT_HASH
DIAGNOSTICS_VERSION=6f323fe
CONFIG_VERSION=900aeed
PKTFWD_VERSION=8aaee4a
MINER_VERSION=32253cf
UPNP_VERSION=e6aae2b
```

and then within gen_docker_compose we check if value of a setting starts with $ then we read it from `os.environ`. This will need to be documented though. The downside is what happens if in the future we want to have another source for reading values apart from environment? I guess we can use value like `$ENV.FIRMWARE_SHORT_HASH`.

### 2. Expose `os.environ` as a built-in to jinja2 template

We don't add `FIRMWARE_SHORT_HASH`  in `settings.ini` but instead add `os.environ` dict as a built-in for template args. Something like:

```python
.......
template_args['$ENV'] = os.environ
output = template.render(**template_args)
```

Then we can access any environment variable within the template without needing to expose it manually. If this does not have any security implications (environment during build in workflow should be pretty lean) we can use this approach.

Overall the aim is that we can have documented built-ins available to the template and apart from that values from `settings.ini` are available.

Reply:
If using the second approach, template would look like:

```
- FIRMWARE_SHORT_HASH={{$ENV['FIRMWARE_SHORT_HASH']}}
```

Reply:
I had dropped the first idea to keep than change minimum and prevent confusion for further releases. And I don't think the second approach is a good idea. Because not all of the `template_args` are coming from the environment. Almost all of them comes from `settings.ini`. So we'll be mixing two different logic here and have to merge environment vars and setting.ini vars in a single dictionary. Also in the template all of the variables has to be converted to ENV.xxx format and I think this would become much more complicated than initially anticipated.

Current style changes only a simple line and adds the variable in the template only in necessary places.

Reply:
The template happens to be actually like this:
```
- FIRMWARE_SHORT_HASH={{ENV.FIRMWARE_SHORT_HASH}}
```
Thanks for the suggestion. It would work for all future changes easily.